### ISSUE TITLE
🐛: Main thread is blocked on biosimulation.org/projects 

### ISSUE THREAD

**Issue by bilalshaikh42** [reproducible-biomedical-modeling/CRBM-Viz#4445]
### Location

BioSimulations

### What happened?

When loading https://biosimulations.org/projects the main thread is blocked.

### What made it happen?

Load https://biosimulations.org/projects
While projects are loading, try to click on the nav buttons. The input is not processed indicating that the main thread is blocked. 

### Version

Production (Deployed on a .org domain)

### What browsers are you seeing the problem on?

Chrome

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by jonrkarr** [comment]
This is one of the performance issues that we've discussed. This needs to be addressed by pagination and/or moving the calculation of filters to the server.

**Comment by bilalshaikh42** [comment]
Yup, I had considered this a slightly different issue since  It might be possible to make the calculation asynchronous in some way. While the loading might still take time, it should be on a separate thread

**Comment by jonrkarr** [comment]
I see. You're right that the calculation could be moved to a separate thread.

**Comment by bilalshaikh42** [comment]
Before working on this,  we should actually profile the page loading. I suspect its more likely caused by waiting for a network request on the main thread than computation. 

But if it is computation, this would be the approach:  https://angular.io/guide/web-worker

**Comment by bilalshaikh42** [comment]
And a quick example: https://dev.to/ngconf/supercharge-your-angular-application-using-web-workers-5h9b
